### PR TITLE
Update Essentials Compass API Docs

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change

Moves the external comments to inline and minor improvements.

### Issues Fixed

Related to #3960

### CODE REVIEW

File: src/Essentials/src/Compass/Compass.shared.cs [hunk 1]
```diff
@@ -95,46 +163,82 @@ public static void SetShouldDisplayHeadingCalibration(this ICompass compass, boo
...
+		/// <param name="other">Object to compare with.</param>
+		/// <returns><see langword="true"/> if they are equal, otherwise <see langword="false"/>.</returns>
 		public bool Equals(CompassData other) =>
 			HeadingMagneticNorth.Equals(other.HeadingMagneticNorth);
 
+		/// <summary>
+		///	Equality operator for equals.
+		/// </summary>
+		/// <param name="left">Left to compare.</param>
+		/// <param name="right">Left to compare.</param>
```
Comment (Redth) [inline src/Essentials/src/Compass/Compass.shared.cs:221]:
```suggestion
		/// <param name="right">Right to compare.</param>
```

File: src/Essentials/src/Compass/Compass.shared.cs [hunk 2]
```diff
@@ -95,46 +163,82 @@ public static void SetShouldDisplayHeadingCalibration(this ICompass compass, boo
...
+		/// <returns><see langword="true"/> if objects are equal, otherwise <see langword="false"/>.</returns>
 		public static bool operator ==(CompassData left, CompassData right) =>
 			left.Equals(right);
 
+		/// <summary>
+		/// Inequality operator.
+		/// </summary>
+		/// <param name="left">Left to compare.</param>
+		/// <param name="right">Left to compare.</param>
```
Comment (Redth) [inline src/Essentials/src/Compass/Compass.shared.cs:230]:
```suggestion
		/// <param name="right">Right to compare.</param>
```

File: src/Essentials/src/Compass/Compass.shared.cs [hunk 3]
```diff
@@ -4,57 +4,115 @@
...
 
+		/// <summary>
+		/// Gets a value indicating whether reading the compass is supported on this device.
+		/// </summary>
 		public static bool IsSupported
 			=> Current.IsSupported;
 
-		/// <include file="../../docs/Microsoft.Maui.Essentials/Compass.xml" path="//Member[@MemberName='IsMonitoring']/Docs/*" />
+		/// <summary>
+		/// Gets if compass is actively being monitored.
```
Comment (hartez) [inline src/Essentials/src/Compass/Compass.shared.cs:77]:
```suggestion
		/// Gets a value indicating whether the compass is actively being monitored.
```

File: src/Essentials/src/Compass/Compass.shared.cs [hunk 4]
```diff
@@ -4,57 +4,115 @@
...
-		/// <include file="../../docs/Microsoft.Maui.Essentials/Compass.xml" path="//Member[@MemberName='Stop'][1]/Docs/*" />
+		/// <summary>
+		/// Stop monitoring for changes to the compass.
+		/// </summary>
 		public static void Stop()
 			=> Current.Stop();
 
 #if IOS || MACCATALYST
+		/// <summary>
+		/// Gets or sets if the calibration screen should be displayed.
```
Comment (hartez) [inline src/Essentials/src/Compass/Compass.shared.cs:113]:
```suggestion
		/// Gets or sets a value specifying whether the calibration screen should be displayed.
```

File: src/Essentials/src/Compass/Compass.shared.cs [hunk 5]
```diff
@@ -75,16 +133,26 @@ public static bool ShouldDisplayHeadingCalibration
...
 	}
 
+	/// <summary>
+	/// /// This class contains static extension methods for use with <see cref="ICompass"/>.
+	/// </summary>
 	public static class CompassExtensions
 	{
 #if IOS || MACCATALYST
+		/// <summary>
+		/// Gets or sets if the calibration screen should be displayed.
```
Comment (hartez) [inline src/Essentials/src/Compass/Compass.shared.cs:153]:
```suggestion
		/// Gets or sets a value specifying whether the calibration screen should be displayed.
```

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Essentials/src/Compass/Compass.shared.cs (hunk 1 above)
Comment (jfversluis) [reply src/Essentials/src/Compass/Compass.shared.cs:221]: 🤦‍♂️

**Review by hartez — APPROVED** [review]

**Pull request merged**